enterprise/backend/src/metabase_enterprise/scim/api.clj

```diff
@@ -2,17 +2,46 @@
   "/api/ee/scim/ endpoints"
   (:require
    [compojure.core :refer [POST]]
+   [metabase-enterprise.serialization.v2.backfill-ids :as serdes.backfill]
    [metabase.api.common :as api :refer [defendpoint]]
    [metabase.models.api-key :as api-key]
+   [metabase.models.setting :as setting :refer [defsetting]]
+   [metabase.public-settings :as public-settings]
+   [metabase.util.i18n :refer [deferred-tru]]
    [metabase.util.secret :as u.secret]
    [toucan2.core :as t2]))
 
 (set! *warn-on-reflection* true)
 
+(defsetting scim-enabled
+  (deferred-tru "Is SCIM currently enabled?")
+  :visibility :admin
+  :type       :boolean
+  :audit      :getter
+  :default    false
+  ;; TODO: should this be exported?
+  :export?    false)
```

Comment:
I think the frontend is supposed to set this value, but could there be an issue because the two requests needed to start using scim (1. changing this setting to true + 2. requesting an api key) might not always happen together? e.g. if the client dies mid request?

If so I think setting scim-enabled to true in the `POST "/api_key"` endpoint (perhaps in a transaction) would fix it. But, I may be missing something there

Reply:
Yeah @sloansparger and I just discussed this. I'll be updating this PR a bit